File: kiosk/php/GetDatabases.php
<?php
/*  Prototype Dreamtech Kiosk, version 1.0.0.0
 *  (c) 2018-2019 Seyoung Park.
 *
 *
 *
 */
//$object = new mssql_db;
//$object->Connect_Databases();
echo $_POST["method"]();

//$row_count = sqlsrv_num_rows( $stmt );
//echo "Row count result = $row_count\n";

//$field_count = sqlsrv_num_fields( $stmt );
//echo "Field count result = $row_count\n";

//class mssql_db{
//    public $conn;
//    public $msg;
//    public $name;

    /*************************************************************************
    NAME
    DESCRIPTION
    RETURNS
    */
    function Connect_Databases()
    {
        if(isset($_POST['server'])) {
            $server = json_decode($_POST['server']);
        }
        if(isset($_POST['username'])) {
            $username = json_decode($_POST['username']);
        }
        if(isset($_POST['password'])) {
            $password = json_decode($_POST['password']);
        }
        if(isset($_POST['portnumber'])) {
            $portnumber = json_decode($_POST['portnumber']);
        }
        if(isset($_POST['dbname'])) {
            $dbname = json_decode($_POST['dbname']);
        }

        $connectionTimeoutSeconds = 15;  // Default of 15 seconds
        $serverName = $server;
        $connectionOptions = array(
            "Database"=>$dbname,
            "UID"=>$username,
            "PWD"=>$password,
            "CharacterSet" => "UTF-8",
            "LoginTimeout" => $connectionTimeoutSeconds);

        //debug_to_console($connectionOptions);
        //sqlsrv_configure('WarningsReturnAsErrors', true);
        //sqlsrv_configure('LogSubsystems', SQLSRV_LOG_SYSTEM_ALL);
        //sqlsrv_configure('LogSeverity', SQLSRV_LOG_SEVERITY_ALL);

        $conn = sqlsrv_connect( $serverName, $connectionOptions);
        return $conn;
    }

    /*************************************************************************
    NAME
    DESCRIPTION
    RETURNS
    */
    function Disconnect_Databases()
    {
        sqlsrv_free_stmt($stmt);
        sqlsrv_close( $conn);
    }

    /*************************************************************************
    NAME
    DESCRIPTION
    RETURNS
    */
    function Get_DB_cal_res(){
        $return = new stdClass;

        // Connection DB
        if(($conn = Connect_Databases()) == true ) {
            $return->success = true;
        }else{
            $return->success = false;
        }

        if($conn == true) {
            // primary key : schedule_seq
            $sql = "SELECT * FROM dbo.cal_res";
            $params = array();
            $options = array(
                //"Scrollable" => 'static'
            );
            $stmt = sqlsrv_query($conn, $sql, $params, $options);
            if( $stmt === false ) {
                $return->success = false;
            }
            else{
                $databaseNames = [
                    'resource_seq' => [],
                    'info' => [],
                    'position' => []
                ];

                while( $row = sqlsrv_fetch_array( $stmt) ) {
                    array_push($databaseNames['resource_seq'], $row['resource_seq']);
                    array_push($databaseNames['info'], $row['info']);
                    array_push($databaseNames['position'], $row['position']);
                }
                $return->success = true;
            }
        }
        $return->errorMessage = sqlsrv_errors();
        $return->data['database'] = $databaseNames;
        $json = json_encode($return);
        Disconnect_Databases();
        echo $json;
    }

    /*************************************************************************
    NAME
    DESCRIPTION
    RETURNS
    */
    function Get_DB_cal_res_sch(){
        $return = new stdClass;

        // Connection DB
        if(($conn = Connect_Databases()) == true ) {
            $return->success = true;
        }else{
            $return->success = false;
        }

        if(isset($_POST['sdate'])) {
            $sdate = json_decode($_POST['sdate']);
        }
        if(isset($_POST['edate'])) {
            $edate = json_decode($_POST['edate']);
        }
        /*
        if(isset($_POST['res_seq'])) {
            $res_seq = json_decode($_POST['res_seq']);
        }
        if(isset($_POST['schedule_seq'])) {
            $schedule_seq = json_decode($_POST['schedule_seq']);
        }
        */
        if($conn == true) {
            //$sql = "SELECT * FROM dbo.cal_res_sch WHERE resource_seq = $res_seq and schedule_seq = $schedule_seq";
            $sql = "SELECT * FROM dbo.cal_res_sch WHERE sdate BETWEEN '$sdate' and '$edate' ORDER BY sdate asc";
            $params = array();
            $options = array(
                /*"Scrollable" => 'static'*/
            );

            $stmt = sqlsrv_query($conn, $sql, $params, $options);
            if( $stmt === false ) {
                $return->success = false;
            }
            else{
                $databaseNames = [
                    'schedule_seq' => [],
                    'resource_seq' => [],
                    'actor' => [],
                    'sdate' => [],
                    'edate' => [],
                    'body' => []
                ];

                while( $row = sqlsrv_fetch_array( $stmt) ) {
                    array_push($databaseNames['schedule_seq'], $row['schedule_seq']);
                    array_push($databaseNames['resource_seq'], $row['resource_seq']);
                    array_push($databaseNames['actor'], $row['actor']);
                    array_push($databaseNames['sdate'], $row['sdate']);
                    array_push($databaseNames['edate'], $row['edate']);
                    array_push($databaseNames['body'], $row['body']);
                    //$sdate_string = date_format( $row['sdate'], 'jS, F Y' );
                    //$edate_string = date_format( $row['edate'], 'jS, F Y' );
                    //$repeat_end_date_string = date_format( $row['repeat_end_date'], 'jS, F Y' );
                }
                $return->success = true;
            }
        }
        $return->errorMessage = sqlsrv_errors();
        $return->data['database'] = $databaseNames;
        $json = json_encode($return);
        Disconnect_Databases();
        echo $json;
    }

    /*************************************************************************
    NAME
    DESCRIPTION
    RETURNS
    */
    function Get_DB_cal_res_view(){
        $return = new stdClass;

        // Connection DB
        if(($conn = Connect_Databases()) == true ) {
            $return->success = true;
        }else{
            $return->success = false;
        }

        if(isset($_POST['sdate'])) {
            $sdate = json_decode($_POST['sdate']);
        }
        if(isset($_POST['edate'])) {
            $edate = json_decode($_POST['edate']);
        }

        if($conn == true)
        {
            // primary key : view_start_date BETWEEN '2018-10-31 00:30:00.000' and '2018-10-31 23:30:00' ORDER BY view_start_date asc
            $sql = "SELECT * FROM dbo.cal_res_view WHERE view_start_date BETWEEN '$sdate' and '$edate' ORDER BY view_start_date asc";
            $params = array();
            $options = array(
                //"Scrollable" => 'static'
            );
            $stmt = sqlsrv_query($conn, $sql, $params, $options);
            if( $stmt === false ) {
                $return->success = false;
            }
            else{
                $databaseNames = [
                    'resource_view_seq' => [],
                    'schedule_seq' => [],
                    'view_start_date' => [],
                    'view_end_date' => []
                ];

                while( $row = sqlsrv_fetch_array( $stmt) ) {
                    array_push($databaseNames['resource_view_seq'], $row['resource_view_seq']);
                    array_push($databaseNames['schedule_seq'], $row['schedule_seq']);
                    array_push($databaseNames['view_start_date'], $row['view_start_date']);
                    array_push($databaseNames['view_end_date'], $row['view_end_date']);
                }
                $return->success = true;
            }
        }
        $return->errorMessage = sqlsrv_errors();
        $return->data['database'] = $databaseNames;
        $json = json_encode($return);
        Disconnect_Databases();
        echo $json;
    }
//}
?>

File: kiosk/py/GetDatabases.py
#!/usr/bin/python
#  Prototype Dreamtech Kiosk, version 1.0.0.0
#  (c) 2018-2019 Seyoung Park.
#
#
#
#
import sys
import os
import json
import pyodbc
import cgi
import app
import collections
import flask
import flask_cors

from flask import Flask, jsonify, render_template, request
from flask_cors import CORS
from datetime import datetime
from json import dumps

app = Flask(__name__)
CORS(app)

conn = None
rMsg = ''
server = '192.168.1.200'
database = 'WO2011'
username = 'dreamtech01'
password = 'dreamtech01'
port = '1433'

"""
    NAME
    DESCRIPTION
    RETURNS
"""
def Disconnect_Databases(con):
    #print('Disconnect_Databases')
    con.close()
    return

"""
    NAME
    DESCRIPTION
    RETURNS
"""
@app.route('/Get_DB_cal_res', methods = ['POST', 'GET'])
def Get_DB_cal_res():
    #print('Get_DB_cal_res')

    ##### create dictionary #####
    objects_list = {'data':{}, 'errorMessage':None, 'success':True}
    objects_list['data'].setdefault('database', {})

    ##### connect mssql #####
    try:
        if request.form['run'] == 'true':
            # for windows ODBC Driver 17 for SQL Server
            info = 'DRIVER={ODBC Driver 17 for SQL Server};SERVER='+request.form['server'].replace("\"", "")+';PORT='+request.form['portnum'].replace("\"", "")+';DATABASE='+request.form['dbname'].replace("\"", "")+';UID='+request.form['username'].replace("\"", "")+';PWD='+ request.form['password'].replace("\"", "")

        else:
            # for raspberrypi FreeTDS Server
            info = 'DRIVER={FreeTDS};SERVER='+request.form['server'].replace("\"", "")+';PORT='+request.form['portnum'].replace("\"", "")+';DATABASE='+request.form['dbname'].replace("\"", "")+';UID='+request.form['username'].replace("\"", "")+';PWD='+ request.form['password'].replace("\"", "")

        conn = pyodbc.connect(info)

    except pyodbc.Error as ex:
        objects_list['errorMessage'] = ex.args[1]
        objects_list['success'] = False

    ##### check the result #####
    if (objects_list['success'] == True):
        cursor = conn.cursor()
        sql_text = 'select * from dbo.cal_res'
        cursor.execute(sql_text)
        result = cursor.fetchall()

        objects_list['data']['database'].setdefault('resource_seq', [])
        objects_list['data']['database'].setdefault('info', [])
        objects_list['data']['database'].setdefault('position', [])

        for row in result:
            objects_list['data']['database']['resource_seq'].append(row.resource_seq)
            objects_list['data']['database']['info'].append(row.info)
            objects_list['data']['database']['position'].append(row.position)

        objects_list['success'] = True
        j = json.dumps(objects_list)
        Disconnect_Databases(conn)
        return j
    else:
        objects_list['success'] = False
        j = json.dumps(objects_list)
        return j

"""
    NAME
    DESCRIPTION
    RETURNS
"""
@app.route('/Get_DB_cal_res_sch', methods = ['POST', 'GET'])
def Get_DB_cal_res_sch():
    #print('Get_DB_cal_res_sch')

    ##### create dictionary #####
    objects_list = {'data':{}, 'errorMessage':None, 'success':True}
    objects_list['data'].setdefault('database', {})

    ##### connect mssql #####
    try:
        if request.form['run'] == 'true':
            # for windows ODBC Driver 17 for SQL Server
            info = 'DRIVER={ODBC Driver 17 for SQL Server};SERVER='+request.form['server'].replace("\"", "")+';PORT='+request.form['portnum'].replace("\"", "")+';DATABASE='+request.form['dbname'].replace("\"", "")+';UID='+request.form['username'].replace("\"", "")+';PWD='+ request.form['password'].replace("\"", "")

        else:
            # for raspberrypi FreeTDS Server
            info = 'DRIVER={FreeTDS};SERVER='+request.form['server'].replace("\"", "")+';PORT='+request.form['portnum'].replace("\"", "")+';DATABASE='+request.form['dbname'].replace("\"", "")+';UID='+request.form['username'].replace("\"", "")+';PWD='+ request.form['password'].replace("\"", "")

        conn = pyodbc.connect(info)

    except pyodbc.Error as ex:
        objects_list['errorMessage'] = ex.args[1]
        objects_list['success'] = False

    ##### check the result #####
    if (objects_list['success'] == True):
        cursor = conn.cursor()

        #print(request.form['sdate'])
        #print(request.form['edate'])

        '''
        d['sdate'] = request.form['sdate']
        d['edate'] = request.form['edate']
        '''
        # $sql = "SELECT * FROM dbo.cal_res_sch WHERE sdate BETWEEN '$sdate' and '$edate' ORDER BY sdate asc";
        # sql_text = 'select * from dbo.cal_res'
        sql_text = 'select * from dbo.cal_res_sch where sdate between '+'\''+request.form['sdate'].replace("\"", "")+'\''+' and '+'\''+request.form['edate'].replace("\"", "")+'\''+' order by sdate asc'

        #print(sql_text)

        cursor.execute(sql_text)
        result = cursor.fetchall()

        objects_list['data']['database'].setdefault('schedule_seq', [])
        objects_list['data']['database'].setdefault('resource_seq', [])
        objects_list['data']['database'].setdefault('actor', [])
        objects_list['data']['database'].setdefault('sdate', [])
        objects_list['data']['database'].setdefault('edate', [])
        objects_list['data']['database'].setdefault('body', [])

        for row in result:
            objects_list['data']['database']['schedule_seq'].append(row.schedule_seq)
            objects_list['data']['database']['resource_seq'].append(row.resource_seq)
            objects_list['data']['database']['actor'].append(row.actor)
            objects_list['data']['database']['sdate'].append(row.sdate.strftime("%Y-%m-%d %H:%M:%S"))
            objects_list['data']['database']['edate'].append(row.edate.strftime("%Y-%m-%d %H:%M:%S"))
            objects_list['data']['database']['body'].append(row.body)

        objects_list['success'] = True
        j = json.dumps(objects_list)
        Disconnect_Databases(conn)
        return j
    else:
        objects_list['success'] = False
        j = json.dumps(objects_list)
        return j

"""
    NAME
    DESCRIPTION
    RETURNS
"""
@app.route('/Get_DB_cal_res_view', methods = ['POST', 'GET'])
def Get_DB_cal_res_view():
    #print('Get_DB_cal_res_view')

    ##### create dictionary #####
    objects_list = {'data':{}, 'errorMessage':None, 'success':True}
    objects_list['data'].setdefault('database', {})

    ##### connect mssql #####
    try:
        if request.form['run'] == 'true':
            # for windows ODBC Driver 17 for SQL Server
            info = 'DRIVER={ODBC Driver 17 for SQL Server};SERVER='+request.form['server'].replace("\"", "")+';PORT='+request.form['portnum'].replace("\"", "")+';DATABASE='+request.form['dbname'].replace("\"", "")+';UID='+request.form['username'].replace("\"", "")+';PWD='+ request.form['password'].replace("\"", "")
        else:
            # for raspberrypi FreeTDS Server
            info = 'DRIVER={FreeTDS};SERVER='+request.form['server'].replace("\"", "")+';PORT='+request.form['portnum'].replace("\"", "")+';DATABASE='+request.form['dbname'].replace("\"", "")+';UID='+request.form['username'].replace("\"", "")+';PWD='+ request.form['password'].replace("\"", "")

        conn = pyodbc.connect(info)

    except pyodbc.Error as ex:
        objects_list['errorMessage'] = ex.args[1]
        objects_list['success'] = False

    ##### check the result #####
    if (objects_list['success'] == True):
        cursor = conn.cursor()

        #print(request.form['sdate'])
        #print(request.form['edate'])

        '''
        d['sdate'] = request.form['sdate']
        d['edate'] = request.form['edate']
        '''

        # $sql = "SELECT * FROM dbo.cal_res_view WHERE view_start_date BETWEEN '$sdate' and '$edate' ORDER BY view_start_date asc";
        # sql_text = 'select * from dbo.cal_res'

        sql_text = 'select * from dbo.cal_res_view where view_start_date between '+'\''+request.form['sdate'].replace("\"", "")+'\''+' and '+'\''+request.form['edate'].replace("\"", "")+'\''+' order by view_start_date asc'

        #print(sql_text)

        cursor.execute(sql_text)
        result = cursor.fetchall()

        objects_list['data']['database'].setdefault('resource_view_seq', [])
        objects_list['data']['database'].setdefault('schedule_seq', [])
        objects_list['data']['database'].setdefault('view_start_date', [])
        objects_list['data']['database'].setdefault('view_end_date', [])

        for row in result:
            objects_list['data']['database']['resource_view_seq'].append(row.resource_view_seq)
            objects_list['data']['database']['schedule_seq'].append(row.schedule_seq)
            objects_list['data']['database']['view_start_date'].append(row.view_start_date.strftime("%Y-%m-%d %H:%M:%S"))
            objects_list['data']['database']['view_end_date'].append(row.view_end_date.strftime("%Y-%m-%d %H:%M:%S"))

        objects_list['success'] = True
        j = json.dumps(objects_list)
        Disconnect_Databases(conn)
        return j
    else:
        objects_list['success'] = False
        j = json.dumps(objects_list)
        return j

"""
    NAME
    DESCRIPTION
    RETURNS
"""
if __name__ == '__main__':
    app.run(debug=True)


'''

conn = Connect_Databases(request.form['server'], request.form['portnum'], request.form['dbname'], request.form['username'], request.form['password'])

#'DRIVER=FreeTDS;SERVER=192.168.1.200;PORT=1433;DATABASE=WO2011;UID=dreamtech01;PWD=dreamtech01;TDS_Version=4.2;'
#'DRIVER={ODBC Driver 17 for SQL Server};SERVER=192.168.1.200;PORT=1433;DATABASE=WO2011;UID=dreamtech01;PWD=dreamtech01;TDS_Version=4.2;'

# for raspberrypi FreeTDS Server
#conn = pyodbc.connect('DRIVER={FreeTDS};SERVER='+server+';PORT='+port+';DATABASE='+database+';UID='+username+';PWD='+ password)

objects_list = []
d = collections.OrderedDict()
d['method'] = request.form['method']
d['server'] = request.form['server']
d['username'] = request.form['username']
d['password'] = request.form['password']
d['portnum'] = request.form['portnum']
d['dbname'] = request.form['dbname']
d['sdate'] = request.form['sdate']
d['edate'] = request.form['edate']
d['res_seq'] = request.form['res_seq']
d['schedule_seq'] = request.form['schedule_seq']
objects_list.append(d)
j = json.dumps(objects_list)
return j '''


'''
@app.route('/Get_DB_cal_res', methods = ['POST', 'GET'])
@app.route('/Get_DB_cal_res_sch', methods = ['POST', 'GET'])
@app.route('/Get_DB_cal_res_view', methods = ['POST', 'GET'])
'''